torchmetrics/detection/map.py

```diff
@@ -329,12 +354,16 @@ def update(self, preds: List[Dict[str, Tensor]], target: List[Dict[str, Tensor]]
             self.detection_boxes.append(boxes)
```

Comment:
no need to store those states if calculating only for masks. This will blow up memory!